servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -335,6 +337,56 @@ public EntriesResponse getEntries(String namedRef, EntriesParams params)
     }
   }
 
+  @Override
+  public EntriesResponse getNamespaceEntries(
+      String refName,
+      @Nullable String hashOnRef,
+      @Nullable ContentsKey namespacePrefix,
+      @Nullable Integer depth)
+      throws NessieNotFoundException {
+    WithHash<NamedRef> refWithHash = namedRefWithHashOrThrow(refName, hashOnRef);
+    try {
+      List<EntriesResponse.Entry> entries;
+      try (Stream<EntriesResponse.Entry> keyStream =
+          getStore()
+              .getKeys(refWithHash.getHash())
+              .map(
+                  key ->
+                      EntriesResponse.Entry.builder()
+                          .name(fromKey(key.getValue()))
+                          .type((Type) key.getType())
+                          .build())) {
+        entries =
+            filterEntries(
+                    keyStream,
+                    (namespacePrefix == null || namespacePrefix.getElements().isEmpty())
+                        ? null
+                        : buildRegex(namespacePrefix))
```

Comment:
A. Do we have to restrict to custom prefix use-case only? Why not take regex as an input from the ReST API instead.
B. Is it possible to pushdown the filter to the underlying store/DB adapter (alternatively)?

Reply:
A. Ive been thinking that we should probably integrate this w/ the `getEntries` api endpoint. We would only have to add a depth parameter to the API then. Filtering would only happen if depth > 0.

Thoughts? cc @snazy @dimas-b

Reply:
B. that would be a goal for a later PR

Reply:
A: from my POV it makes sense to add an optional `depth` parameter to `getEntries`.

I'm not sure about exposing a RegEx filter to the end user is worth the effort. I suppose the main use case is hierarchical traversal, so depth alone should be sufficient. Depth should also be easier to push down to storage level  than a generic RegEx, I guess.